app/javascript/javascript/singleSession.js

```diff
@@ -5,21 +5,27 @@
 // The same for the empty point at the end.
 export const measurementsToTimeWithExtremes = ({ measurements, times }) => {
   const { start, end } = times;
-  return {
+  const ms = {
     [start]: {
       x: start,
       y: null,
       latitude: null,
       longitude: null
     },
-    ...measurementsToTime(measurements),
-    [end]: {
-      x: end,
-      y: null,
-      latitude: null,
-      longitude: null
-    }
+    ...measurementsToTime(measurements)
   };
+
+  return ms.hasOwnProperty(end)
```

Comment:
for example this in my opinion doesn't read well, just by looking at this line I don't understand what's going on

Reply:
What would you suggest?

Reply:
it's just that if `ms` had a longer more descriptive way then this line would read better. don't know what `ms` should be instead tho without analyzing the other code around.

Reply:
Makes sense! Unfortunately, I'm deep into a project and cannot take care of that. But feel free to tackle it if you feel like it!